src/fourier.jl

```diff
@@ -0,0 +1,781 @@
+    cross /= n_ave
+    unnorm_cross /= n_ave
+    if return_auxil
+        unnorm_pds1 /= n_ave
+        unnorm_pds2 /= n_ave
+    end
+
+    # Finally, normalize the cross spectrum (only if not already done on an
+    # interval-to-interval basis)
+    if use_common_mean
+        cross = normalize_periodograms(
+            unnorm_cross,
+            dt,
+            n_bin;
+            mean_flux = common_mean,
+            n_ph=n_ph,
+            norm=norm,
+            variance=common_variance,
+            power_type=power_type,
+        )
+        if return_auxil
+            pds1 = normalize_periodograms(
+                unnorm_pds1,
+                dt,
+                n_bin;
+                mean_flux = common_mean1,
+                n_ph=n_ph1,
+                norm=norm,
+                variance=common_variance1,
+                power_type=power_type,
+            )
+            pds2 = normalize_periodograms(
+                unnorm_pds2,
+                dt,
+                n_bin;
+                mean_flux = common_mean2,
+                n_ph=n_ph2,
+                norm=norm,
+                variance=common_variance2,
+                power_type=power_type,
+            )
+        end
+    end
+    # If the user does not want negative frequencies, don't give them
+    if !(fullspec)
+        freq = freq[fgt0]
+    end
+
+    results = DataFrame()
+    results[!,"freq"] = freq
+    results[!,"power"] = cross
+    results[!,"unnorm_power"] = unnorm_cross
+
+    if return_auxil
+        results[!,"pds1"] = pds1
+        results[!,"pds2"] = pds2
+        results[!,"unnorm_pds1"] = unnorm_pds1
+        results[!,"unnorm_pds2"] = unnorm_pds2
+    end
+
+    results = attach_metadata(results,(n= n_bin, m= n_ave, dt= dt,
+                         norm= norm,
+                         df= 1 / (dt * n_bin),
+                         segment_size= dt * n_bin,
+                         nphots= n_ph,
+                         nphots1= n_ph1, nphots2= n_ph2,
+                         countrate1= common_mean1 / dt,
+                         countrate2= common_mean2 / dt,
+                         mean= common_mean,
+                         mean1= common_mean1,
+                         mean2= common_mean2,
+                         power_type= power_type,
+                         fullspec= fullspec,
+                         variance= common_variance,
+                         variance1= common_variance1,
+                         variance2= common_variance2))
+
+    return results
+    
+end
+
+function avg_pds_from_events(times:: AbstractVector{<:Real}, gti::AbstractMatrix{<:Real}, 
+                             segment_size::Real, dt::Real; norm::String="frac",
+                             use_common_mean::Bool=true, silent::Bool=false, 
+                             fluxes=nothing, errors=nothing)
+    if isnothing(segment_size)
+        segment_size = max(gti) - min(gti)
+    end
+    n_bin = Int64(round(segment_size / dt))
```

Comment:
```suggestion
    n_bin = segment_size ÷ dt
```

Reply:
Well Integer Division always rounds down right?

Reply:
Good point, then the first suggestion in https://github.com/matteobachetti/Stingray.jl/pull/2#discussion_r917182140 stands.